libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/src/azure/core/az_base64.c

```diff
@@ -440,12 +440,14 @@ az_base64_decode(az_span destination_bytes, az_span source_base64_text, int32_t*
     source_index += 4;
   }
 
-  // We are guaranteed to have an input with at least 4 bytes at this point, with a size that is a
-  // multiple of 4.
-  int32_t i0 = *(source_ptr + source_length - 4);
-  int32_t i1 = *(source_ptr + source_length - 3);
-  int32_t i2 = *(source_ptr + source_length - 2);
-  int32_t i3 = *(source_ptr + source_length - 1);
+  // If length is divisible by four, do nothing. Else, we assume up to two padding characters.
+  int32_t source_length_mod_four = source_length % 4;
```

Comment:
Looks like a bug in the Base64 implementation from Azure SDK.
I think it would be useful to have a tracking issue there as well.

Reply:
The padding assumption or the calculations below?